api/src/graphql/LearnedSkill.ts

```diff
@@ -0,0 +1,111 @@
+import { Skill } from "@prisma/client";
+import { extendType, objectType } from "nexus";
+import { Context } from "../context";
+
+export const LearnedSkill = objectType({
+  name: "LearnedSkill",
+  definition(t) {
+    t.nonNull.int("count");
+    t.nonNull.field("skill", { type: "Skill" })
+  },
+});
+
+interface countObject {
+  [key: number]: number
+}
+
+interface learnedSkillsCountObject {
+  count: number;
+  skill: Skill;
+}
+
+// 集計処理
+// skill idをkey, intをvalueとして走査し、incrementする
+function countSkillsUsedOnPosts(skillsOfAllPosts: Skill[][]):countObject {
+  const counts: countObject = {};
+  for (let i = 0; i < skillsOfAllPosts.length; ++i) {
+    const skillsOfEachPost = skillsOfAllPosts[i];
+    for (let j = 0; j < skillsOfEachPost.length; ++j) {
+      const skill = skillsOfEachPost[j];
+      const id: number = skill['id'];
```

Comment:
maybe more readable? its up to you
```suggestion
      const id: number = skill.id;
```